### DISCUSSION TITLE
Apps from casks won't start after updating (without further action)

### DISCUSSION THREAD

**Discussion by michael1006** [Homebrew/discussions#3171] [Casks]
I tried filing this as an [issue](https://github.com/Homebrew/homebrew-autoupdate/issues/86) but didn't get a response, so thought I'd ask here instead -- or someone please let me know if I should file it as an issue under /brew rather than /homebrew-autoupdate:

I used homebrew to install most of my Macs apps as casks, e.g. apps like Google Chrome. A few months ago I installed homebrew on a brand-new Macbook Air M1 with Monterey with apps.

The problem I have is that, having set apps to autoupdate every 24h, every time an app actually _does_ autoupdate, it won't automatically restart. The old version of the app quits, but the updated version won't start automatically.

This happens with lots of always-running apps that live in my menubar like Stats, but it also happens whenever I (infrequently) restart Chrome. Today, for example, I put "chrome://restart" in its menubar, and while it quit, it didn't restart. And then when I went to launch Chrome from its icon, I got this warning:

> Homebrew Cask downloaded this file today at 15:20 from www.google.com. Apple checked it for malicious software and none was detected.

Then I clicked OK and Chrome started. So the issue is clearly Gatekeeper -- when apps are supposed to auto-restart they don't (and doesn't even show a prompt), while with manually started apps it does show a prompt.

Is there a way to bypass the prompt and have apps start without one? I don't want to turn off Gatekeeper entirely, and the funny thing is that Gatekeeper explicitly reports it _doesn't_ find malicious software, so I don't even understand why there's a prompt in the first place.

This has sadly been getting tiresome (there are menubar apps I expect to be always-on, and not die on every update), and what I don't understand is that back when I use apps own auto-update feature, this is never a problem. I never was asked to approve each Chrome upgrade by Gatekeeper until I started using the cask autoupdate instead of Chrome's own.

So my questions are:

1) Is this a problem other people are having?

2) Is there a way for me to fix it on my Mac I don't know about? (Without entirely disabling Gatekeeper)

3) Is there a way for brew (auto)update to not suffer from this problem?

I'm actually most mystified because this doesn't appear to be a common problem -- Googing the text of the Gatekeeper message together with homebrew doesn't seem to yield much.

Thank you!

**Comment by toobuntu** [comment]
1. Yes, others do experience this. E.g., discussion #3104.
2. There appear to be two issues here: restarting certain apps that quit during their upgrade and Gatekeeper.
    - **Restart**: See [my comments](https://github.com/orgs/Homebrew/discussions/3104#discussioncomment-7901584) in the referenced discussion or how I do it in [Babble](https://github.com/toobuntu/babble).
    - **Gatekeeper**: Use the `--no-quarantine` option with `brew upgrade`. See the man page for more details. This can be optionally also be set in the environment variable `HOMEBREW_CASK_OPTS`.
3. Yes, add the functionality of 2a to Homebrew Autoupdate and [ensure](https://github.com/Homebrew/homebrew-autoupdate/issues/97) you are using `--no-quarantine` with Homebrew Autoupdate.

**Reply by SMillerDev** [reply]
Passing that flag will disable most of the security features of the casks so don't take that decision to pass it lightly

**Reply by toobuntu** [reply]
Very true. One should consider the relevant factors when deciding _whether_ to disable Gatekeeper for an app. For upgrades only, I assume the decision was already made to trust the app and its developer. At least with Homebrew, most casks have checksums to verify the download is what's expected.

One could additionally pass `--require-sha` (or add it to `HOMEBREW_CASK_OPTS`) to require that all casks have a checksum. Though if you have an app without a checksum and want to upgrade it automatically, that option would interfere.

> [Gatekeeper](https://support.apple.com/en-us/HT202491) ... is designed to make sure that only trusted software runs on your Mac.
> Gatekeeper performs online checks to verify if an app contains known malware and whether the developer’s signing certificate is revoked.

> [Gatekeeper](https://support.apple.com/en-ca/guide/security/sec5599b66df/web) also protects against the distribution of malicious plug-ins with benign apps.
